src/sentry/metrics/minimetrics.py

```diff
@@ -25,17 +25,22 @@ def patch_sentry_sdk():
     real_add = MetricsAggregator.add
     real_emit = MetricsAggregator._emit
 
-    @wraps(real_add)
-    def tracked_add(self, ty, *args, **kwargs):
-        real_add(self, ty, *args, **kwargs)
+    @metrics_noop
+    def report_tracked_add(ty):
         metrics.incr(
             key="minimetrics.add",
             amount=1,
             tags={"metric_type": ty},
             sample_rate=1.0,
         )
 
+    @wraps(real_add)
+    def tracked_add(self, ty, *args, **kwargs):
+        real_add(self, ty, *args, **kwargs)
+        report_tracked_add(ty)
+
     @wraps(real_emit)
+    @metrics_noop
```

Comment:
This is the actual fix #2.

Reply:
Note here: i want to add a change to the python SDK which protects the manual `flush` call. Once that lands, the `real_emit` must not be marked as `metrics_noop` or it stops being called. The unittests should catch this though.